common/src/test/kotlin/streams/service/sink/strategy/CUDIngestionStrategyTest.kt

```diff
@@ -0,0 +1,572 @@
+        val mergeNodeFooBar = findEventByQuery("""
+                |${StreamsUtils.UNWIND}
+                |MERGE (n:Foo:Bar {key: event.${CUDIngestionStrategy.ID_KEY}.key})
+                |SET n += event.properties
+            """.trimMargin(), nodeEvents)
+        assertEquals(1, mergeNodeFooBar.events.size)
+        assertNodeEventsContainsKey(mergeNodeFooBar, key)
+        val mergeNodeFooBarLabel = findEventByQuery("""
+                |${StreamsUtils.UNWIND}
+                |MERGE (n:Foo:Bar:Label {key: event.${CUDIngestionStrategy.ID_KEY}.key})
+                |SET n += event.properties
+            """.trimMargin(), nodeEvents)
+        assertEquals(2, mergeNodeFooBarLabel.events.size)
+        assertNodeEventsContainsKey(mergeNodeFooBarLabel, key)
+        val updateNodeFooBar = findEventByQuery("""
+                |${StreamsUtils.UNWIND}
+                |MATCH (n:Foo:Bar {key: event.${CUDIngestionStrategy.ID_KEY}.key})
+                |SET n += event.properties
+            """.trimMargin(), nodeEvents)
+        assertEquals(1, updateNodeFooBar.events.size)
+        assertNodeEventsContainsKey(updateNodeFooBar, key)
+        val updateNodeFooBarLabel = findEventByQuery("""
+                |${StreamsUtils.UNWIND}
+                |MATCH (n:Foo:Bar:Label {key: event.${CUDIngestionStrategy.ID_KEY}.key})
+                |SET n += event.properties
+            """.trimMargin(), nodeEvents)
+        assertEquals(1, updateNodeFooBarLabel.events.size)
+        assertNodeEventsContainsKey(updateNodeFooBarLabel, key)
+    }
+
+    @Test
+    fun `should delete nodes`() {
+        // given
+        val detachMarkers = listOf(1, 3, 8, 10)
+        val list = (1..10).map {
+            val labels = if (it % 2 == 0) listOf("Foo", "Bar") else listOf("Foo", "Bar", "Label")
+            val detach = it in detachMarkers
+            val properties = mapOf("foo" to "foo-value-$it", "id" to it)
+            val cudNode = CUDNode(op = CUDOperations.delete,
+                    labels = labels,
+                    ids = mapOf("key" to it),
+                    properties = properties,
+                    detach = detach)
+            StreamsSinkEntity(null, cudNode)
+        }
+
+        // when
+        val cudQueryStrategy = CUDIngestionStrategy()
+        val nodeEvents = cudQueryStrategy.mergeNodeEvents(list)
+        val nodeDeleteEvents = cudQueryStrategy.deleteNodeEvents(list)
+
+        val relationshipEvents = cudQueryStrategy.mergeRelationshipEvents(list)
+        val relationshipDeleteEvents = cudQueryStrategy.deleteRelationshipEvents(list)
+
+        // then
+        assertEquals(emptyList(), nodeEvents)
+        assertEquals(emptyList(), relationshipEvents)
+        assertEquals(emptyList(), relationshipDeleteEvents)
+
+        assertEquals(4, nodeDeleteEvents.size)
+        assertEquals(10, nodeDeleteEvents.map { it.events.size }.sum())
+        val deleteNodeFooBar = findEventByQuery("""
```

Comment:
Recommend testing the deletion by a compound key (>= 2 properties).

Reply:
Ok